src/components/WalletConnect/NanoContract/NewNanoContractTransactionRequest.js

```diff
@@ -0,0 +1,313 @@
+import FeedbackModal from '../../FeedbackModal';
+import errorIcon from '../../../assets/images/icErrorBig.png';
+import checkIcon from '../../../assets/images/icCheckBig.png';
+import { DappContainer } from './DappContainer';
+import { NanoContractExecInfo } from './NanoContractExecInfo';
+import { NanoContractActions } from './NanoContractActions';
+import { NanoContractMethodArgs } from './NanoContractMethodArgs';
+import { DeclineModal } from './DeclineModal';
+
+/**
+ * @param {Object} props
+ * @param {Object} props.ncTxRequest
+ * @param {Object} props.ncTxRequest.nc
+ * @param {Object} props.ncTxRequest.dapp
+ * @param {string} props.ncTxRequest.dapp.icon
+ * @param {string} props.ncTxRequest.dapp.proposer
+ * @param {string} props.ncTxRequest.dapp.url
+ * @param {string} props.ncTxRequest.dapp.description
+ */
+export const NewNanoContractTransactionRequest = ({ ncTxRequest }) => {
+  const { nc, dapp } = ncTxRequest;
+  const dispatch = useDispatch();
+  const navigation = useNavigation();
+  const newTxStatus = useSelector((state) => state.walletConnect.newNanoContractTransaction.status);
+  const firstAddress = useSelector((state) => state.firstAddress);
+  // Nullable if the nano contract method is 'initialize'
+  const registeredNc = useSelector((state) => state.nanoContract.registered[nc.ncId]);
+  const registeredTokensMetadata = useSelector((state) => state.tokenMetadata);
+  // Use it to add loading feedback
+  const metadataLoaded = useSelector((state) => state.metadataLoaded);
+  const blueprintInfo = useSelector((state) => state.nanoContract.blueprint[nc.blueprintId]);
+
+  const [showSelectAddressModal, setShowSelectAddressModal] = useState(false);
+  const [showDeclineModal, setShowDeclineModal] = useState(false);
+  // If nano-contract's method is 'initialize' then the expression
+  // should be resolved to firstAddress value by default.
+  // However, the wallet can have trouble to
+  const [ncAddress, setNcAddress] = useState(registeredNc?.address || firstAddress.address);
+  const ncToAccept = useMemo(() => ({
+    ...nc,
+    caller: ncAddress,
+  }), [ncAddress])
+
+  // Controle SelectAddressModal
+  const toggleSelectAddressModal = () => setShowSelectAddressModal(!showSelectAddressModal);
+  const handleAddressSelection = (newAddress) => {
+    setNcAddress(newAddress);
+    toggleSelectAddressModal();
+  };
+
+  // Accepts the Nano Contract data preseted.
+  const onAcceptTransaction = () => {
+    // Update the caller with the address selected by the user.
+    const acceptedNc = { ...nc, caller: ncAddress };
+    // Signal the user has accepted the current request and pass the accepted data.
+    dispatch(walletConnectAccept(acceptedNc));
+  };
+
+  const onDeclineTransaction = () => {
+    setShowDeclineModal(true);
+  };
+  const onDeclineConfirmation = () => {
+    setShowDeclineModal(false);
+    dispatch(walletWalletReject());
+    navigation.goBack();
+    // TODO: It is not being dimissed automatically
```

Comment:
I refactored this in my PR, so leave this as a TODO

Reply:
Done.